window_manager/src/MsWindow.hpp

```diff
@@ -0,0 +1,135 @@
+#ifndef RDMN_MS_WINDOW_HPP
+#define RDMN_MS_WINDOW_HPP
+
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+#include <atomic>
+#include <future>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <type_traits>
+
+#include <rcbe-engine/datamodel/rendering/RenderingContext.hpp>
+#include <rcbe-engine/datamodel/system/WindowContext.hpp>
+#include <rcbe-engine/datamodel/system/window_config.hpp>
+#include <rcbe-engine/core/AbstractInputManager.hpp>
+#include <rcbe-engine/core/windows_input_event.hpp>
+#include <rcbe-engine/renderer/Renderer.hpp>
+
+#include <rdmn-engine/logger/trivial_logger.hpp>
+
+namespace rcbe::core {
+template <typename Implementation>
+class BaseWindow {
+public:
+	static LRESULT CALLBACK windowLoopHandler(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam) {
+		Implementation* impl_this = nullptr;
+
+		if (msg == WM_NCCREATE) {
+			CREATESTRUCT* create_struct_ptr = reinterpret_cast<CREATESTRUCT*>(lparam);
+			impl_this = static_cast<Implementation*>(create_struct_ptr->lpCreateParams);
+			SetWindowLongPtr(hwnd, GWLP_USERDATA, reinterpret_cast<LONG_PTR>(impl_this));
+
+			impl_this->setWindowHandle(hwnd);
+		} else {
+			impl_this = reinterpret_cast<Implementation*>(GetWindowLongPtr(hwnd, GWLP_USERDATA));
+		}
+
+		if (impl_this) {
+			rdmn::core::windows_input_event we{.type = msg, .wparam = wparam, .lparam = lparam};
+			//return impl_this->handleMessage(msg, wparam, lparam);
+			return impl_this->handleMessage(we);
+		} else {
+			return DefWindowProc(hwnd, msg, wparam, lparam);
+		}
+	}
+
+	bool create(
+		PCWSTR lpWindowName,
+		DWORD dwStyle,
+		DWORD dwExStyle = 0,
```

Comment:
Leave some comment explaining what is this.

Reply:
Done